### PR TITLE
Port axum 0.2 upgrade to 0.11.x branch

### LINKED ISSUE #738: Release new `askama_axum`? Existing latest release doesn't pass "cargo audit".
Our project fails `cargo audit` due to:

```
       > Version:       0.1.2
       > Title:         No default limit put on request bodies
       > Date:          2022-08-31
       > ID:            RUSTSEC-2022-0055
       > URL:           https://rustsec.org/advisories/RUSTSEC-2022-0055
       > Solution:      Upgrade to >=0.2.8, <0.3.0-rc.1 OR >=0.3.0-rc.2
       > Dependency tree:
       > axum-core 0.1.2
       >
       > error: 1 vulnerability found!
```

and `askama_axum` is pulling in this dependency. I see that the version in `git` already points to `axum-core = "0.2"`, so that would help, but needs a release.

Thanks!

### PR DESCRIPTION
**Pull request by djc**
Fixes #738.

### CODE REVIEW

File: askama_shared/src/generator.rs [hunk 1]
```diff
@@ -221,7 +221,7 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         buf.writeln("#[inline]")?;
         buf.writeln(
             "fn into_response(self)\
-             -> ::askama_axum::Response<::askama_axum::BoxBody> {",
+             -> ::askama_axum::Response {",
```
Comment (Kijewski) [inline askama_shared/src/generator.rs:224]: The linebreak can be omitted.

**Review by Kijewski — CHANGES_REQUESTED** [review]
Please rebase. Except for the one comment it looks good to me.

I love the simplification, thank you @jplatte!

**Comment by djc** [comment]
Not sure what you meant by rebase? This intentionally targets the 0.11.x branch.

**Comment by dnaka91** [comment]
Axum just released a new version, should this be bumped to the now new `axum-core` v`0.3.0` instead?

**Comment by djc** [comment]
Let's merge this as is and then someone can follow up with a 0.3 PR.

**Review by Kijewski — APPROVED** [review]
LGTM

**Comment by dnaka91** [comment]
Seems a bump to 0.3 is already happening in #716 🎉

**Pull request merged**